File: src/main/scala/com/vgk/vgkmod/CustomTeleporter.scala
package com.vgk.vgkmod
import net.minecraft.world.{WorldServer, Teleporter}
import net.minecraft.entity.Entity

class CustomTeleporter(world: WorldServer, x: Double, y: Double, z: Double) extends Teleporter(world) {

  override def placeInPortal(entity: Entity, rotationYaw: Float) {
    entity.setPosition(this.x, this.y, this.z)
    System.out.println("Hello")
    entity.motionX = 0.0f
    entity.motionY = 0.0f
    entity.motionZ = 0.0f
  }
}

File: src/main/java/com/vgk/vgkmod/CustomTeleporter.java
package com.vgk.vgkmod;

import javax.annotation.Nonnull;

import net.minecraft.entity.Entity;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.Teleporter;
import net.minecraft.world.WorldServer;

public class CustomTeleporter extends Teleporter {
    private final double x, y, z;
    
    @SuppressWarnings("all")
    public CustomTeleporter(WorldServer world, double x, double y, double z) {
        super(world);
        this.x = x;
        this.y = y;
        this.z = x;
    }
    

    @Override
    public void placeInPortal(@Nonnull Entity entity, float rotationYaw) {
        entity.setPosition(this.x, this.y, this.z);
        entity.motionX = 0.0f;
        entity.motionY = 0.0f;
        entity.motionZ = 0.0f;
    }
}